enterprise/server/cmd/ci_runner/main.go

```diff
@@ -452,16 +551,16 @@ func (ar *actionRunner) Run(ctx context.Context, ws *workspace, startTime time.T
 		if err != nil {
 			return status.InvalidArgumentErrorf("failed to parse bazel command: %s", err)
 		}
-		printCommandLine(ar.log, bazeliskBinaryName, args...)
+		printCommandLine(ar.reporter.log, bazeliskBinaryName, args...)
 		// Transparently set the invocation ID from the one we computed ahead of
 		// time. The UI is expecting this invocation ID so that it can render a
 		// BuildBuddy invocation URL for each bazel_command that is executed.
 		args = append(args, fmt.Sprintf("--invocation_id=%s", iid))
 
-		runErr := runCommand(ctx, *bazelCommand, args, nil /*=env*/, ar.log)
+		runErr := runCommand(ctx, *bazelCommand, args, nil /*=env*/, ar.reporter.log)
 		exitCode := getExitCode(runErr)
 		if exitCode != noExitCode {
-			ar.log.Printf("%s(command exited with code %d)%s\n", ansiGray, exitCode, ansiReset)
+			ar.reporter.log.Printf("%s(command exited with code %d)%s\n", ansiGray, exitCode, ansiReset)
```

Comment:
not really necessary to fix rn, and i know it wasn't your doing in this CL, but it's a little bit of a code smell to be reaching so deep into another objects internal state (ar.reporter.log.Fn()) -- my main worry with code like this is accidentally hitting a NPE, but in general i think it's a sign that some logic should be moved from the callsite where this is happening into the object itself

Reply:
I changed buildEventReporter to implement the logging methods so callers can use the reporter as a logger directly.